### PR TITLE
PA-22 array flags replace all server side regexps at query time, also readme improvements and minor fixes

### PR DESCRIPTION
**Pull request by boutell**

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -156,14 +156,7 @@ module.exports = async function(plants) {
...
         if (!filter.choices) {
-          const raw = (await plants.distinct(filter.name)).filter(choice => ((typeof choice) === 'string') && choice.length);
-          if (filter.commaSeparated) {
-            filter.choices = [... new Set(raw.map(choice => choice.split(/\s*,+\s*/)).flat()) ];
-          } else if (filter.commaOrSemicolonSeparated) {
-            filter.choices = [... new Set(raw.map(choice => choice.split(/\s*(?:,|;)+\s*/)).flat()) ];
-          } else {
-            filter.choices = raw;
-          }
+          filter.choices = (await plants.distinct(filter.name)).filter(choice => ((typeof choice) === 'string') && choice.length);
```
Comment (boutell) [inline lib/server.js:159]: All the heavy processing moved to the massage script where it's one and done

File: massage.js [hunk 2]
```diff
@@ -121,14 +121,30 @@ async function go() {
...
     });
+    const soilMoistureFlags = plant['Soil Moisture'].split(', ').map(capitalize).filter(flag => flag.length > 0);
+    await plants.updateOne({
+      _id: plant._id
+    }, {
+      $set: {
+        'Soil Moisture Flags': soilMoistureFlags
+      }
+    });
+    const pollinatorFlags = plant['Pollinators'].split(/\s*(?:,|;)+\s*/).map(capitalize).filter(flag => flag.length > 0);
```
Comment (boutell) [inline massage.js:140]: Break out all the distinct values into a nice array

File: ui/src/App.vue [hunk 3]
```diff
@@ -85,20 +85,22 @@ export default {
           }
         },
         {
-          name: 'Soil Moisture',
-          commaSeparated: true,
-          value: []
+          name: 'Soil Moisture Flags',
```
Comment (boutell) [inline ui/src/App.vue:88]: Browser side obeys the same schema of fields as the server side

**Review by ellatenar — APPROVED** [review]
Looks good!

**Pull request merged**